modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -446,6 +471,9 @@ private void requestPartitions(final RebalanceFuture fut, GridDhtPreloaderAssign
                 if (fut.isDone())
                     break;
 
+                if (rebalanceFut.rebalancingStartTime == -1)
```

Comment:
Do we need such check?

Reply:
Without this check, the rebalancing start time can be set several times, because it is in the “for each” cycle